dask/tests/test_base.py

```diff
@@ -342,6 +342,51 @@ def __dask_graph__(self):
     assert not is_dask_collection(DummyCollection)
 
 
+def test_unpack_collections():
+    a = delayed(1) + 5
+    b = a + 1
+    c = a + 2
+
+    def build(a, b, c, iterator):
+        return (a, b,               # Top-level collections
+                {'a': a,            # dict
+                 a: b,              # collections as keys
+                 'b': [1, 2, b],    # list
```

Comment:
Can I ask for a nested list, perhaps just by subbing `b -> [b]`

Reply:
Done.